Calibration/HcalCalibAlgos/macros/CalibMonitor.C

```diff
@@ -240,24 +253,28 @@ CalibMonitor::CalibMonitor(std::string fname, std::string dirnm,
   if (plotType_ < 0 || plotType_ > 3) plotType_ = 3;
   flexibleSelect_  = (((flag_/1) %10));
   corrPU_          = (((flag_/1000) %10) > 0);
+  selRBX_          = (((flag_/10000) %10) > 0);
+  coarseBin_       = (((flag_/100000) %10) > 0);
   log2by18_        = std::log(2.5)/18.0;
   TFile      *file = new TFile(fname.c_str());
   TDirectory *dir  = (TDirectory*)file->FindObjectAny(dirnm.c_str());
   std::cout << fname << " file " << file << " " << dirnm << " " << dir 
 	    << " flags " << flexibleSelect_ << "|" << plotType_ << "|"
 	    << corrPU_ << " cons " << log2by18_ << " eta range " << etalo_ 
 	    << ":" << etahi_ << " run range " << runlo_ << ":" << runhi_ 
-	    << std::endl;
+	    << " Selection of RBX " << selRBX_ << std::endl;
   TTree      *tree = (TTree*)dir->Get("CalibTree");
   std::cout << "CalibMonitor:Tree " << tree << std::endl;
   Init(tree,dupFileName);
   corrE_ = ReadCorrFactor(corrFileName);
+  if (rcorFileName != "") cFactor_ = new CalibCorr(rcorFileName,false);
   std::cout << "Reads correction factors from " << corrFileName << " with flag "
 	    << corrE_ << std::endl;
 }
 
 CalibMonitor::~CalibMonitor() {
-  if (!fChain) return;
+  if (cFactor_) delete cFactor_;
+  if (!fChain)  return;
```

Comment:
if your pointers are `nullptr` you can still `delete` them, in case of `nullptr`s just nothing happens